thorlcr/master/thdemonserver.cpp

```diff
@@ -152,6 +153,8 @@ class DeMonServer : public CSimpleInterface, implements IDeMonServer
                     CGraphBase &graph = activeGraphs.item(g2);
                     unsigned startTime = graphStarts.item(g2);
                     reportStatus(wu, graph, startTime, finished, success);
+                    // Prune subgraph descendant stats as they have been serialized and no longer needed.
+                    globalStatsCollection.pruneSubGraphDescendants(wfid, graphName, graph.queryGraphId());
```

Comment:
Also, whilst true these aren't needed in globalStatsCollection, they will be repopulated repeatedly on every call to reportActiveGraphs.

I think it would better to clear once, in endGraph()